cli/src/helpers/dynamic_format.rs

```diff
@@ -0,0 +1,101 @@
+// Copyright (C) 2019-2023 Aleo Systems Inc.
+// This file is part of the snarkOS library.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::{
+    atomic::{AtomicBool, Ordering},
+    Arc,
+};
+
+use time::format_description::{self, BorrowedFormatItem};
+use tracing::{Event, Subscriber};
+use tracing_subscriber::{
+    fmt::{format::Writer, FmtContext, FormatEvent, FormatFields},
+    registry::LookupSpan,
+};
+
+pub struct DynamicFormatter<'b> {
+    dim_format: DimFormat<'b>,
+    default_format: tracing_subscriber::fmt::format::Format,
+    dim: Arc<AtomicBool>,
```

Comment:
I presume that this is the same atomic as the shutdown flag? It would be a good idea to comment on that so that it's clear.